nodebuilder/header/header.go

```diff
@@ -0,0 +1,148 @@
+package header
+
+import (
+	"context"
+	"encoding/hex"
+
+	"github.com/ipfs/go-datastore"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/libp2p/go-libp2p-core/peerstore"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	"github.com/multiformats/go-multiaddr"
+	tmbytes "github.com/tendermint/tendermint/libs/bytes"
+
+	"github.com/celestiaorg/celestia-node/fraud"
+	"github.com/celestiaorg/celestia-node/header"
+	"github.com/celestiaorg/celestia-node/header/p2p"
+	"github.com/celestiaorg/celestia-node/header/store"
+	"github.com/celestiaorg/celestia-node/header/sync"
+	"github.com/celestiaorg/celestia-node/params"
+	headerservice "github.com/celestiaorg/celestia-node/service/header"
+)
+
+// Service creates a new header.Service.
+func Service(
+	syncer *sync.Syncer,
+	sub header.Subscriber,
+	p2pServer *p2p.ExchangeServer,
+	ex header.Exchange,
+	store header.Store,
+) *headerservice.Service {
+	return headerservice.NewHeaderService(syncer, sub, p2pServer, ex, store)
+}
+
+// P2PExchange constructs new Exchange for headers.
+func P2PExchange(cfg Config) func(params.Bootstrappers, host.Host) (header.Exchange, error) {
+	return func(bpeers params.Bootstrappers, host host.Host) (header.Exchange, error) {
+		peers, err := cfg.trustedPeers(bpeers)
+		if err != nil {
+			return nil, err
+		}
+		ids := make([]peer.ID, len(peers))
+		for index, peer := range peers {
+			ids[index] = peer.ID
+			host.Peerstore().AddAddrs(peer.ID, peer.Addrs, peerstore.PermanentAddrTTL)
+		}
+		return p2p.NewExchange(host, ids), nil
+	}
+}
+
+// InitStore initializes the store.
+func InitStore(cfg *Config) func(context.Context, params.Network, header.Store, header.Exchange) error {
```

Comment:
Does this have to return a function? Or can it just be invoked here directly and just return an error?